packages/wallet/src/services/DelegationTracker/RewardAccounts.ts

```diff
@@ -196,8 +196,32 @@ const accountCertificateTransactions = (
   );
 };
 
+const accountDRepCertificateTransactions = (
+  transactions$: Observable<TxWithEpoch[]>,
+  rewardAccount: Cardano.RewardAccount
+) => {
+  const stakeKeyHash = Cardano.RewardAccount.toHash(rewardAccount);
+  return transactions$.pipe(
+    map((transactions) =>
+      transactions
+        .map(({ tx, epoch }) => ({
+          certificates: (tx.body.certificates || [])
+            .map((cert) =>
+              Cardano.isCertType(cert, [...Cardano.VoteDelegationCredentialCertificateTypes]) ? cert : null
+            )
+            .filter(isNotNil)
+            .filter((cert) => (cert.stakeCredential.hash as unknown as Crypto.Ed25519KeyHashHex) === stakeKeyHash),
```

Comment:
Shouldn't we check if the voting delegation certificate is younger than the latest stake key registration certificate?
I mean, if we delegate our vote then we unregister and re-register our stake credentials, is the vote delegated as early than we unregistered the stake credential or is it in an _un-delegated_ status?

I understand that checking for stake de-registration certificates is not required: if a stake credential is no longer registered, it has no rewards to withdrawn, none of these checks will be performed.

Reply:
Good catch @iccicci , You are right, if we un-register and re-register the key, it returns to an un-delegated state, will add this check

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1466/commits/829fd1f8fedc4bc1a2bf3702c95c8f318a9b731a